README.md

```diff
@@ -1,122 +1,318 @@
-# Final Project Starter
+# AI CV Generator (Full Stack App)
 
-## Overview
+This is a full-stack resume/CV builder app that lets users create AI-enhanced resumes by filling out a multi-step form. The app integrates with the **Gemini API (Google AI)** to generate polished CV content tailored to job criteria.
 
-This repository contains a starter template for your final project, including both a React frontend and an Express backend. The project is designed to be modular, scalable, and easy to maintain.
-
-To get started with this template just use the [Github UI](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template) to make a copy and get coding 🚀
+---
 
 ## Project Structure
 
-```
-final-project-starter/
-├── web/                  # React frontend application
-├── api/                  # Express backend API
-└── package.json          # Global scripts and dependencies
+src/
+├── components/
+│ ├── Button/
+│ ├── CharacterCount/
+│ ├── Education/
+│ ├── ErrorState/
+│ ├── Header/
+│ ├── IconSlide/
+│ ├── LeftPane/
+│ ├── LoadingState/
+│ ├── MultiFormPage/
+│ ├── PersonalInfo/
+│ ├── Playground/
+│ ├── ProfessionalSummary/
+│ ├── ProfileVsJob/
+│ ├── Project/
+│ └── TransferableExperience/
+├── hooks/
+│ └── useSubmitPersonalInfo.js
+├── utils/
+│ └── (frontend helper utils - no validation)
+├── pages/
+api/
+├── controllers/
+│ ├── generateCv.js
+│ └── enhanceWithAi.js
+├── routes/
+│ ├── aiEnhanceRoute.js
+│ └── cvRoute.js
+├── utils/
+│ └── validations.js ← Contains only API key validation logic
+├── server.js
+└── .env
+
+---
+
+## Features
+
+### Frontend (React)
+
+- Gemini API key authentication
+- Multi-step form with auto-save to `localStorage`
+- Real-time controlled inputs for:
+  - Personal Info
+  - Professional Summary
+  - Transferable Experience
+  - Education
+  - Projects
+  - Profile vs Job Matching
+- AI-enhanced CV generation
+- Sidebar with progress tracking
+
+### Backend (Node.js + Express)
+
+- Validates form input using `yup`
+- Validates Google Gemini API key format
+- Handles Gemini AI requests for CV enhancement
+- Returns structured CV data for frontend display/download
+
+---
+
+## How It Works (User Flow)
+
+- The user enters a Gemini API key to unlock the CV form
+- The key is validated and saved to `localStorage`
+- User fills out a multi-step form guided by a LeftPane and IconSlider
+- Progress is **auto-saved** to avoid data loss
+- On form submission, data is validated and sent to the backend
+- Gemini AI processes the CV data and returns an enhanced version
+- Enhanced CV is displayed/downloaded
```

Comment:
I think we should split this into two points.
- CV is displayed with editable functionality to allow the user to change and expand upon the AI generation
- A PDF is downloadable ready to apply to jobs